### ISSUE TITLE
.execSingle()/.limit(1).exec() vs .exec() - different values

### ISSUE THREAD

**Issue by VuongN** [shiftcode/dynamo-easy#253]
One of my current queries on a GSI is giving me odd results:

[QUERY]
`
    const account = await AccountStore
        .query()
        .index(EMAIL_INDEX)
        .wherePartitionKey(username)
        .whereAttribute("dataType").equals(constants.DATA_TYPE_ACCOUNT)
        .exec();
`

`[QUERY].execSingle()/.limit(1).exec()` => `null`
`[QUERY].exec() => `[{item}]`

They have the same exact query. Did I hit a weird bug or am I missing something crucial here?

Thank you!
-V.

**Comment by simonmumenthaler** [comment]
Hi @VuongN 
You use a where condition. By using limit `limit(1)`/`execSingle()` (which also uses Limit=1), dynamoDB reads exactly the first item in this partition. If this item does not match your condition --> null

Query without limit looks through the whole partition and returns all items matching your condition (but it actually read all of them).

it's the way dynamoDB works :)

**Comment by VuongN** [comment]
Fair enough. For some reason I was expecting a little magic, but no-magic is the best decision here. Thank you.
